sw/host/tests/crypto/ecdh_kat/src/main.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use arrayvec::ArrayVec;
+use clap::Parser;
+use num_bigint_dig::BigUint;
+use std::fs;
+use std::time::Duration;
+
+use serde::Deserialize;
+
+use cryptotest_commands::commands::CryptotestCommand;
+use cryptotest_commands::ecdh_commands::{
+    CryptotestEcdhCoordinate, CryptotestEcdhCurve, CryptotestEcdhDeriveOutput,
+    CryptotestEcdhOperation, CryptotestEcdhPrivateKey, EcdhSubcommand,
+};
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::rpc::{UartRecv, UartSend};
+use opentitanlib::uart::console::UartConsole;
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    // Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "10s")]
+    timeout: Duration,
+
+    #[arg(long, num_args = 1..)]
+    ecdh_json: Vec<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct EcdhTestCase {
+    algorithm: String,
+    operation: String,
+    curve: String,
+    private: BigUint,
+    shared: BigUint,
+    qx: BigUint,
+    qy: BigUint,
+    result: bool,
+}
+
+const ECDH_CMD_MAX_PRIVATE_KEY_BYTES: usize = 32;
+const ECDH_CMD_MAX_COORDINATE_BYTES: usize = 33;
```

Comment:
Why 33 here instead of 32?

Reply:
Obsoleted. I was originally confused about the test vectors' integer encoding when developing this.